lib/ssh/setup-ssh-agent

```diff
@@ -0,0 +1,9 @@
+sshdir=$(getent passwd $USER | cut -d: -f6)/.ssh
+mkdir $sshdir
+chmod 700 $sshdir
+cat > $sshdir/key
+chmod 600 $sshdir/key
+eval "$(ssh-agent -s)"
```

Comment:
sshdir is allowed to be group-other readable -- people are meant to be able to see your pubkeys. It's only writability that's a concern there.

To write the key without group-other readability even during race conditions:


```suggestion
mkdir -p "$sshdir"
umask 077
cat > "$sshdir/key"
eval "$(ssh-agent)"
```

([-s is default](https://www.man7.org/linux/man-pages/man1/ssh-agent.1.html); `mkdir -p` is all that's needed to make this script idempotent)

It obviously doesn't matter much here, but it's a good habit.

P3 or even P4

Reply:
removed all this in favor of reading the key from stdin.